### DISCUSSION TITLE
Naming Things

### DISCUSSION THREAD

**Discussion by jdhitsolutions** [jdhitsolutions/PSBluesky#2] [General]
Naming commands is as much of an art as anything. You want the command to be discoverable and unique enough to avoid naming collisions. I have been using the `PSBluesky` prefix for the sake of clarity. To be honest, right now, the module has one command you will use, and that is `New-PSBlueskyPost` , which I've aliased to `skeet.`

**Comment by thedavecarroll** [comment]
Great to have an easy PowerShell tool to interact with [Bluesky](https://bsky.app/).

As you said, naming commands is an art and very subjective. And I truly appreciate you asking for the community's input on the names.

Here are my general suggestions:

- Module command prefix names (not to be confused with `Import-Module -Prefix`) should be kept to 3 to 4 characters, if possible.
- Any prefixes should omit `PS`, `PoSh`, and `Pwsh`, as these should be assumed if one is entering a `Verb-Noun` command.
- For Bluesky (officially only the `B` is uppercase), the brand is commonly abbreviated as `Bsky`.

Following these, I think the commands should be name like `Verb-BskyCommand`.

Though I'm slightly less opinionated on the name of the module. For sharing outside the immediate PowerShell community, I think `PSBluesky` would be fine.

**Reply by potatoqualitee** [reply]
Yes, I am also interested as the screenshots looked pretty! But like `Az` in lieu of `Azure`, shorter is much preferred and my vote is strongly in favor of `Bsky`. As for the module name, PSBluesky makes sense.

**Reply by jdhitsolutions** [reply]
I can get behind a shorter name.

```powershell
Get-BskyProfile jdhitsolutions.com
Get-PSBskyProfile jdhitsolutions.com
```

Although I'm on the fence about including `PS` as part of the noun, my reasoning is that the commands use the API but transform the results into PowerShell objects. I'm also a big fan of command aliases. Maybe `BSProfile` for `Get-PSBskyProfile`.

**Reply by markekraus** [reply]
> my reasoning is that the commands use the API but transform the results into PowerShell objects.

Can you elaborate on this? An SDK's primary purpose is to transform API results into something usable for the SDK language. That would be expected, so including PS in the command name is superfluous, IMO. 

edit: To clarify my position, I'm for `PSBluesky` as the module name because that will be great for module discoverability. But I believe `Bsky` should be the command prefix.

**Reply by jdhitsolutions** [reply]
I've renamed commands to `Bsky*`.  The module name and type names will remain unchanged. This will be effective in v0.5.0.

**Comment by jdhitsolutions** [comment]
I appreciate the input. I want to settle this before I publish to the PowerShell Gallery.
